lib/assert.js

```diff
@@ -185,6 +185,36 @@ function isArguments(object) {
   return Object.prototype.toString.call(object) == '[object Arguments]';
 }
 
+function compareMaps(a, b, strict) {
+  // check size of the maps
+  if (a.size !== b.size)
+    return false;
+
+  // iterate over all keys and check if they exist in the other
+  for (let kvs of a) {
```

Comment:
it is not faster, but performs equally in this kind of situation: http://jsperf.com/let-vs-var-iteration
I believe the optimization arrived with V8 4.4.